adagrams/game.py

```diff
@@ -1,11 +1,175 @@
+              "K":1,
+              "L":4,
+              "M":2,
+              "N":6,
+              "O":8,
+              "P":2,
+              "Q":1,
+              "R":6,
+              "S":4,
+              "T":6,
+              "U":4,
+              "V":2,
+              "W":2,
+              "X":1,
+              "Y":2,
+              "Z":1}
+
+SIZE_OF_DRAW = 10
+
+LETTERS_VALUES = {"A":1,
+                  "B":3,
+                  "C":3,
+                  "D":2,
+                  "E":1,
+                  "F":4,
+                  "G":2,
+                  "H":4,
+                  "I":1,
+                  "J":8,
+                  "K":5,
+                  "L":1,
+                  "M":3,
+                  "N":1,
+                  "O":1,
+                  "P":3,
+                  "Q":10,
+                  "R":1,
+                  "S":1,
+                  "T":1,
+                  "U":1,
+                  "V":4,
+                  "W":4,
+                  "X":8,
+                  "Y":4,
+                  "Z":10}
+
+WINNER_LEN = 10
+
+def gen_pool_letters(pool_dict):
+    my_pool = list()
+    for letter,frecuency in pool_dict.items():
+        for i in range(0,frecuency):
+            my_pool.append(letter)
+
+    return my_pool
+
+#def init_game():
+#   POOL = gen_pool_letters(POOL_RULES)
+
 def draw_letters():
-    pass
+    draw = list()
+    my_ran_pool = gen_pool_letters(POOL_RULES)
+    
+    #generate random
+    random.seed()
+    for i in range(0,SIZE_OF_DRAW):
+        ran_num = random.randint(0,len(my_ran_pool)-1)
+        draw.append(my_ran_pool[ran_num])
+        my_ran_pool.pop(ran_num)
+
+    return draw
+
+#creates a dictionary which keys are the elements on the letters array 
+#and the value is their frecuency
+def create_dic_repeatead_letters(letters):
+    letters_dict = dict()
+
+    for letter in letters:
+        if letter in letters_dict.keys():
+            letters_dict[letter] += 1
+        else:
+            letters_dict[letter] = 1
+
+    return letters_dict
 
 def uses_available_letters(word, letter_bank):
-    pass
+    #Validate input
+    if word == "" or letter_bank is None:
+        return False
+
+    #Create a dictionary with the number of each letter of the letter bank
+    letter_bank_dict = create_dic_repeatead_letters(letter_bank)
+
+    #Compare each letter of word with the dictionary and their frecuency
+    for letter in word:
+        up_letter = letter.upper()
+        if up_letter not in letter_bank_dict.keys():
+            return False
+        if letter_bank_dict[up_letter]==0:
+            return False  #The letter has been used more times than allowed
+        else:
+            letter_bank_dict[up_letter]+=-1
+
+    return True
 
 def score_word(word):
-    pass
+    points = 0
+    if word is None:
+        return points
+
+    for letter in word:
+        letter_up = letter.upper()
+        if letter_up not in LETTERS_VALUES.keys():
+            return 0
```

Comment:
Something to consider: If we find any characters in the input `word` that aren't in the scoring list, the whole word will get a value of 0. If we wanted the game to allow words with characters like hyphens, one option could be to skip those characters and take no action rather than returning 0.